src/main/java/vct/col/util/FeatureScanner.java

```diff
@@ -15,106 +14,42 @@
 import vct.col.ast.stmt.composite.Switch;
 import vct.col.ast.stmt.composite.Synchronized;
 import vct.col.ast.stmt.composite.TryCatchBlock;
-import vct.col.ast.stmt.composite.VectorBlock;
 import vct.col.ast.stmt.decl.ASTClass;
 import vct.col.ast.generic.ASTNode;
 import vct.col.ast.stmt.decl.ASTSpecial;
 import vct.col.ast.stmt.decl.Contract;
 import vct.col.ast.stmt.decl.Method;
 import vct.col.ast.stmt.terminal.ReturnStatement;
-import vct.col.ast.type.PrimitiveSort;
 import vct.col.ast.util.RecursiveVisitor;
-import vct.col.ast.type.Type;
 
 public class FeatureScanner extends RecursiveVisitor<Object> {
   public FeatureScanner(){
     super(null,null);
   }
-  
-  private boolean has_parallel_blocks=false;
-  private boolean has_statics=false;
-  private boolean has_dynamics=false;
-  private boolean has_doubles=false;
-  private boolean has_longs=false;
-  private boolean has_processes=false;
-  private boolean has_inheritance=false;
-  private boolean has_kernels=false;
+
   private boolean has_iteration_contracts=false;
   private boolean has_finally = false;
   private boolean has_return = false;
-  private boolean has_switch = false;
-  private boolean has_synchronized_modifier;
-  private boolean has_synchronized_statement = false;
   private boolean has_catch = false;
   private boolean has_throwing_method_calls = false;
-  private boolean uses_csl=false;
   private boolean has_throwing_methods = false;
   private EnumSet<StandardOperator> ops_used=EnumSet.noneOf(StandardOperator.class);
   private EnumSet<ASTSpecial.Kind> specials_used=EnumSet.noneOf(ASTSpecial.Kind.class);
   private EnumSet<Binder> binders_used=EnumSet.noneOf(Binder.class);
-  
-  private HashSet<Class<? extends ASTNode>> nodes=new HashSet<Class<? extends ASTNode>>();
 
   public static FeatureScanner scan(ASTNode node) {
     FeatureScanner fs = new FeatureScanner();
     node.accept(fs);
     return fs;
   }
-  
```

Comment:
@niomaster we are removing big chunks of this because we don't really used them right now but maybe they are still useful in the short term? Or do you agree we should remove them?

Reply:
You could say that we are not removing it because it will always be in the git history. However I will discuss with @niomaster whether it is useful to remove all this code.

Reply:
We're not losing particularly complex logic here, I think it's fine to remove this stuff. In fact, we can probably replace FeatureScanner with FeatureRainbow at some point.